src/components/SingleCluster.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { RouteComponentProps } from 'react-router-dom';
+import { PageSectionVariants } from '@patternfly/react-core';
+import { Cluster, getClusters, handleApiError } from '../api';
+import { ErrorState, LoadingState, PageSection } from './ui';
+import { routeBasePath } from '../config';
+
+type SingleClusterProps = RouteComponentProps;
+
+const SingleCluster: React.FC<SingleClusterProps> = ({ history }) => {
+  const [error, setError] = React.useState('');
+  const [clusters, setClusters] = React.useState<Cluster[]>();
+
+  const fetchClusters = React.useCallback(async () => {
+    try {
+      const { data } = await getClusters();
+      setClusters(data);
+    } catch (e) {
+      return handleApiError(e, () => setError('Failed to fetch cluster.'));
+    }
+  }, [setClusters]);
+
+  React.useEffect(() => {
+    fetchClusters();
+  }, [fetchClusters]);
+
+  React.useEffect(() => {
+    if (clusters) {
```

Comment:
> clusters are always true because it is an array. 

Not in recent implementation. The `useState()` is called without param, so it can be `undefined`.
The `clusters === undefined` is farther used to show loading spinner.

> We don't need to use effect here, instead we could just render Redirect component (either to ~new or clusterId)

The effect fetches cluster list (with a single item) to get the `clusterId`. Otherwise redirects to `~new`. Mind the component is rendered for the `/clusters` path.
I can redirect using `<Redirect>` component, but this will make the logic more complicated compared to the recent use of `history`. Anyway, the fetching needs to be handled in an effect.